src/runtime_src/core/pcie/windows/device_windows.cpp

```diff
@@ -204,8 +204,19 @@ xclbin_fcn(const device_type* device, qr_type qr, const std::type_info&, boost::
     throw std::runtime_error("Query request " + std::to_string(qr) + "requires a userpf device");
 
   if (qr == qr_type::QR_MEM_TOPOLOGY_RAW) {
-    std::vector<char> data(4_k);
-    userpf::get_mem_topology(uhdl, data.data(), 4_k);
+    size_t size_ret = 0;
+    userpf::get_mem_topology(uhdl, nullptr, 0, &size_ret);
+    std::vector<char> data(size_ret);
+    userpf::get_mem_topology(uhdl, data.data(), size_ret, nullptr);
+    value = data;
```

Comment:
I suspect this is an unnecessary copy.  Check if boost::any supports move assignment.

Reply:
ok done